consensus/safety-rules/src/tests/suite.rs

```diff
@@ -485,3 +486,89 @@ fn test_voting_potential_commit_id(func: RoundCallback) {
         a3.block().id(),
     );
 }
+
+fn test_sign_bad_proposals(func: RoundCallback) {
```

Comment:
I think this is new. We shouldn't have an 85 line test, no one can reasonably read this without focus. Tests and code in general should be accessible and intuitive. Can we break this up into one function per underlying test case?